src/libraries/System.Private.CoreLib/src/System/Decimal.DecCalc.cs

```diff
@@ -381,7 +376,7 @@ private static uint Div96By64(ref Buf12 bufNum, ulong den)
 
                 // Compute full remainder, rem = dividend - (quo * divisor).
                 //
-                ulong prod = UInt32x32To64(quo, (uint)den); // quo * lo divisor
+                ulong prod = quo * (den & uint.MaxValue); // quo * lo divisor
```

Comment:
This probably needs a comment on what it's doing.

But, notably, this may regress 32-bit platforms as it will now do a more expensive `64x64=64` multiplication, rather than doing the cheaper `32x32=64`.

In general an internal `Math.BigMul(uint a, uint b, out uint low)` could be defined that users `Bmi2.MultiplyNoFlags`, `ArmBase.MultiplyHigh`, and otherwise falls back to the naive algorithm of `(ulong)a * b`

Reply:
Moved all the stuff into an internal `ulong Math.BigMul(uint, uint)` (since we had a `long Math.BigMul(int, int)`). I tried using initrinsic for x86 on 32 bit. Doesn't seem to exist a `ArmBase.MultiplyHigh` yet.